src/java.base/share/classes/java/math/BigDecimal.java

```diff
@@ -87,14 +87,16 @@
  * representations (with different scales), the rules of arithmetic
  * and rounding must specify both the numerical result and the scale
  * used in the result's representation.
-
- * The different representations of the same numerical value are called
- * members of the same <i>cohort</i>. The {@linkplain
+ *
+ * The different representations of the same numerical value are
+ * called members of the same <i>cohort</i>. The {@linkplain
  * compareTo(BigDecimal) natural order} of {@code BigDecimal}
  * considers members of the same cohort to be equal to each other. In
  * contrast, the {@link equals equals} method requires both the
  * numerical value and representation to be the same for equality to
- * hold.
+ * hold. The results of methods like {@link scale} and {@link
+ * unscaledValue} will differ for numerically equal values with
+ * different representations.
```

Comment:
While this text is correct and reasonable, it doesn't really explain _why_ equals() considers the representation. One might assume incorrectly that the representation is internal only and doesn't affect computations. Of course it does affect computations, which is why I suggested the example. Maybe the example doesn't belong right here, in which case it's reasonable for this change to proceed. I think it would be good to put an example _somewhere_ of non-substitutability of numerical values with different representations. Maybe we could handle that separately.

Reply:
I've filed DK-8261862: "Expand discussion of rationale for BigDecimal equals/compareTo semantics" as a follow-up bug.